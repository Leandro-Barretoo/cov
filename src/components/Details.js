import { useSelector } from 'react-redux';

const Details = () => {
  const infoList = useSelector((state) => state.covidReducer);
  let deaths;
  let confirmed;
  let recovered;
  let critical;

  console.log(window.location.pathname.slice(1));

  const renderInformation = infoList.map((covid) => {
    if (covid.code === window.location.pathname.slice(1)) {
      console.log(covid);
      deaths = covid.latest_data.deaths;
      confirmed = covid.latest_data.confirmed;
      recovered = covid.latest_data.recovered;
      critical = covid.latest_data.critical;
    }
    return 'No Countrie found';
  });

  return (
    { renderInformation },
      <div>
        <div>
          <span>
            Latest data:
          </span>
          <span>
            Confirmed:
            {confirmed}
          </span>
          <span>
            Recovered:
            {recovered}
          </span>
          <span>
            Critical:
            {critical}
          </span>
          <span>
            Deaths:
            {deaths}
          </span>
        </div>
      </div>
  );
};

export default Details;
