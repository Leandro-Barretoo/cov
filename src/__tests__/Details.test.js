import React from 'react';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import Details from '../components/Homepage';
import store from '../redux/configureStore';

describe('Components: Details', () => {
  it('renders without crashing', () => {
    expect(
      shallow(
        <Provider store={store}>
          <Details />
        </Provider>,
      ).length,
    ).toEqual(1);
    expect(
      shallow(
        <Provider store={store}>
          <Details />
        </Provider>,
      ).length,
    ).toMatchSnapshot();
  });
});
