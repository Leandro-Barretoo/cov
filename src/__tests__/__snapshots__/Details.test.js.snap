// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: Details renders without crashing 1`] = `1`;
